src/silo_api/database_directory_watcher.test.cpp

```diff
@@ -0,0 +1,61 @@
+#include "silo_api/database_directory_watcher.h"
+
+#include <gtest/gtest.h>
+
+using silo_api::DatabaseDirectoryWatcher;
+
+TEST(DatabaseDirectoryWatcher, emptyDirectoryNotValid) {
+   ASSERT_EQ(DatabaseDirectoryWatcher::checkValidDataSource(""), std::nullopt);
+}
+
+TEST(DatabaseDirectoryWatcher, invalidDirectoryNotValid) {
+   ASSERT_EQ(DatabaseDirectoryWatcher::checkValidDataSource("///"), std::nullopt);
+}
+
+TEST(DatabaseDirectoryWatcher, validBackwardsCompatible) {
+   auto under_test =
+      DatabaseDirectoryWatcher::checkValidDataSource("testBaseData/testDataDirectories/1234");
+   ASSERT_TRUE(under_test.has_value());
+   ASSERT_TRUE(under_test.value().isCompatibleVersion());
```

Comment:
This will change soon (and thus break the test). Let's check for `== 0` instead?

Reply:
(Same for the test below)